x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/action_connector_form/action_type_menu.test.tsx

```diff
@@ -167,6 +167,71 @@ describe('connector_add_flyout', () => {
     });
   });
 
+  describe('filtering', () => {
+    it('filters connectors by search field correctly', async () => {
+      const onActionTypeChange = jest.fn();
+
+      const actionType1 = actionTypeRegistryMock.createMockActionTypeModel({
+        id: 'action-type-1',
+        iconClass: 'test',
+        selectMessage: 'test',
+        validateParams: (): Promise<GenericValidationResult<unknown>> => {
+          const validationResult = { errors: {} };
+          return Promise.resolve(validationResult);
+        },
+        actionConnectorFields: null,
+      });
+
+      const actionType2 = actionTypeRegistryMock.createMockActionTypeModel({
+        id: 'action-type-2',
+        iconClass: 'test',
+        selectMessage: 'test',
+        validateParams: (): Promise<GenericValidationResult<unknown>> => {
+          const validationResult = { errors: {} };
+          return Promise.resolve(validationResult);
+        },
+        actionConnectorFields: null,
+      });
+
+      loadActionTypes.mockResolvedValue([
+        {
+          id: actionType1.id,
+          enabled: true,
+          name: 'Test1',
+          enabledInConfig: true,
+          enabledInLicense: true,
+          minimumLicenseRequired: 'basic',
+          supportedFeatureIds: ['alerting'],
+        },
+        {
+          id: actionType2.id,
+          enabled: true,
+          name: 'Test2',
+          enabledInConfig: true,
+          enabledInLicense: true,
+          minimumLicenseRequired: 'basic',
+          supportedFeatureIds: ['cases'],
+        },
+      ]);
+      actionTypeRegistry.get.mockReturnValue(actionType1);
+
+      const wrapper = mountWithIntl(
+        <ActionTypeMenu
+          onActionTypeChange={onActionTypeChange}
+          actionTypeRegistry={actionTypeRegistry}
+          searchValue="Test1"
+        />
+      );
+      await act(async () => {
+        await nextTick();
+        wrapper.update();
+      });
+
+      expect(wrapper.find('[data-test-subj="action-type-1-card"]').exists()).toBeTruthy();
+      expect(wrapper.find('[data-test-subj="action-type-2-card"]').exists()).toBeFalsy();
```

Comment:
I think it is better if we test a scenario when only one of the two cards is being shown. With this test, we are not sure if searching worked or if all of the cards are being shown due to a bug. What about changing the `name` and the `selectMessage` to be different for each connector type to be sure that we are filtering correctly?